module/zfs/zfs_vfsops.c

```diff
@@ -453,24 +458,40 @@ zfs_userspace_many(zfs_sb_t *zsb, zfs_userquota_prop_t type,
 	zap_attribute_t za;
 	zfs_useracct_t *buf = vbuf;
 	uint64_t obj;
+	int offset = 0;
 
 	if (!dmu_objset_userspace_present(zsb->z_os))
```

Comment:
It would make sense to add the following to this check.

```
	if ((type == ZFS_PROP_USERUSED || type == ZFS_PROP_GROUPUSED ||
	    type == ZFS_PROP_USERQUOTA || type == ZFS_PROP_GROUPQUOTA) &&
 	    !dmu_objset_userspace_present(zsb->z_os))
```

This kind of check occurs a few times so adding a wrapper function to this check this and the *OBJ* versions would be useful.